src/messaging/tests/echo/.gn

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2020 Project CHIP Authors
```

Comment:
I don't think there is any need for these support files for tests.

- We don't want to have high costs for adding tests, such as copying this project boilerplate.
- It doesn't serve a clear purpose, unlike in the "examples" case. The tests are part of CHIP not separate projects.

Instead add `src/messaging/tests/echo` to `src/BUILD.gn`. Can make a new group() if the current groups aren't a good fit.

Reply:
FWIW I don't think the project stuff should exist in config/. either. Would like to remove that and only have them in examples/, where we are explicitly trying to structure things as if each directory were in an independent project.

Reply:
@mspang Could you provide more specific info on what build construct files need to be removed or rewriten?

Reply:
.gn, args.gni, and symlinks (build, build_overrides, third_party) should be removed

These files should only exist in the root of a project, which this is not.

Reply:
Done